src/test/java/edu/cornell/kfs/fp/document/service/impl/CuDisbursementVoucherTaxServiceImplTest.java

```diff
@@ -0,0 +1,302 @@
+package edu.cornell.kfs.fp.document.service.impl;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.vnd.businessobject.VendorAddress;
+import org.kuali.kfs.vnd.businessobject.VendorContract;
+import org.kuali.kfs.vnd.businessobject.VendorDetail;
+import org.kuali.kfs.vnd.businessobject.VendorHeader;
+import org.kuali.kfs.vnd.businessobject.VendorRoutingComparable;
+import org.kuali.kfs.vnd.document.service.VendorService;
+import org.kuali.rice.core.api.parameter.ParameterEvaluator;
+import org.kuali.rice.core.api.parameter.ParameterEvaluatorService;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.krad.bo.Note;
+import org.kuali.rice.krad.document.Document;
+
+public class CuDisbursementVoucherTaxServiceImplTest{
+	
+	private static final Integer FOREIGN_VENDOR_HEADER_ID = 1000;	
+	private static final Integer NON_FOREIGN_VENDOR_HEADER_ID = 2000;
+	private static final String PAYMENT_REASON_THAT_REQUIRES_TAX_REVIEW_FOR_FOREIGN_VENDOR = "Z";
+	private static final String PAYMENT_REASON_THAT_DOES_NOT_REQUIRE_TAX_REVIEW_FOR_FOREIGN_VENDOR = "F";
+	
+	private CuDisbursementVoucherTaxServiceImpl disbursementVoucherTaxService;
+	
+	@Before
+	public void setUp() throws Exception {
+		disbursementVoucherTaxService = new CuDisbursementVoucherTaxServiceImpl();
+		disbursementVoucherTaxService.setVendorService(new MockVendorService());
+		disbursementVoucherTaxService.setParameterEvaluatorService(new MockParameterEvaluatorService());	
+	}
+	
+	@Test
+	public void testIsForeignVendorAndTaxReviewRequired_true()  {
+		boolean isForeignVendorAndTaxReviewRequired = disbursementVoucherTaxService.isForeignVendorAndTaxReviewRequired(KFSConstants.PaymentPayeeTypes.VENDOR, PAYMENT_REASON_THAT_REQUIRES_TAX_REVIEW_FOR_FOREIGN_VENDOR, FOREIGN_VENDOR_HEADER_ID);
+        assertTrue("isForeignVendorAndTaxReviewRequired method should have returned true when input is foreign vendor and payment reason that requires tax review", isForeignVendorAndTaxReviewRequired);
+    }
+	
+	@Test
+	public void testIsForeignVendorAndTaxReviewRequired_false_PayeeTypeNotVendor()  {
+		boolean isForeignVendorAndTaxReviewRequired = disbursementVoucherTaxService.isForeignVendorAndTaxReviewRequired(KFSConstants.PaymentPayeeTypes.EMPLOYEE, PAYMENT_REASON_THAT_REQUIRES_TAX_REVIEW_FOR_FOREIGN_VENDOR, FOREIGN_VENDOR_HEADER_ID);
```

Comment:
In the actual application, I think this method will get invoked with a null vendor header ID if the payee is not a vendor. Not sure if that needs to be reflected in this test.